pkg/sql/logictest/tests/local/local_test: TestLogic_show_ranges failed

pkg/sql/logictest/tests/local/local_test.TestLogic_show_ranges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20245082?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20245082?buildTab=artifacts#/) on master @ [afef2d239966541de7611883fbb803f08c0fed92](https://github.com/cockroachdb/cockroach/commits/afef2d239966541de7611883fbb803f08c0fed92):


```
=== RUN   TestLogic_show_ranges
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/b5a19b64f368c9c0f5a9324ae3af02c1/logTestLogic_show_ranges1648577819
    test_log_scope.go:82: use -show-logs to present logs inline
[01:37:25] setting distsql_workmem='85860B';
[01:37:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1972/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/show_ranges with config local: 34 tests, 1 failures
[01:37:28] --- total progress: 34 statements
--- total: 34 tests, 1 failures
    logic.go:4550: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b5a19b64f368c9c0f5a9324ae3af02c1/logTestLogic_show_ranges1648577819
--- FAIL: TestLogic_show_ranges (3.01s)
=== RUN   TestLogic_show_ranges/show_ranges_from_database/with_tables
    logic.go:3148: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1972/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/show_ranges:269: SELECT start_key, end_key, range_id, schema_name, table_name, table_id, table_start_key, table_end_key
        FROM [SHOW RANGES WITH TABLES]
        ORDER BY range_id
        expected:
            start_key        end_key          range_id  schema_name  table_name  table_id  table_start_key  table_end_key
            /Table/72        /Table/106/1/10  74        public       t           106       /Table/106       /Table/106/1/10
            /Table/106/1/10  /Table/106/2/20  75        public       t           106       /Table/106/1/10  /Table/106/2/20
            /Table/106/2/20  /Table/106/2/30  76        public       t           106       /Table/106/2/20  /Table/106/2/30
            /Table/106/2/30  /Table/107/1/42  77        public       t           106       /Table/106/2/30  /Table/107
            /Table/106/2/30  /Table/107/1/42  77        public       u           107       /Table/107       /Table/107/1/42
            /Table/107/1/42  /Max             78        public       u           107       /Table/107/1/42  /Table/108
            
        but found (query options: "colnames") :
            start_key        end_key          range_id  schema_name  table_name  table_id  table_start_key  table_end_key
            /Table/72        /Table/106/1/10  74        public       t           106       /Table/106       /Table/106/1/10
            /Table/106/1/10  /Table/106/2/20  75        public       t           106       /Table/106/1/10  /Table/106/2/20
            /Table/106/2/20  /Table/106/2/30  76        public       t           106       /Table/106/2/20  /Table/106/2/30
            /Table/106/2/30  /Table/107/1/42  77        public       u           107       /Table/107       /Table/107/1/42
            /Table/106/2/30  /Table/107/1/42  77        public       t           106       /Table/106/2/30  /Table/107
            /Table/107/1/42  /Max             78        public       u           107       /Table/107/1/42  /Table/108
            
--- FAIL: TestLogic_show_ranges/show_ranges_from_database/with_tables (0.13s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_show_ranges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-52912

--- a/pkg/sql/logictest/testdata/logic_test/show_ranges
+++ b/pkg/sql/logictest/testdata/logic_test/show_ranges
@@ -268,7 +268,7 @@ start_key  end_key  raw_start_key  raw_end_key  range_id  schema_name  table_nam
 query TTITTITT colnames
 SELECT start_key, end_key, range_id, schema_name, table_name, table_id, table_start_key, table_end_key
 FROM [SHOW RANGES WITH TABLES]
-ORDER BY range_id
+ORDER BY range_id, table_id
 ----
 start_key        end_key          range_id  schema_name  table_name  table_id  table_start_key  table_end_key
 /Table/72        /Table/106/1/10  74        public       t           106       /Table/106       /Table/106/1/10

